packages/dashboard/src/layout/useListener.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from 'react';
+import type { EventEmitter } from '@deephaven/golden-layout';
+
+/**
+ * Listen for a specific event on an EventEmitter
+ * @param eventEmitter EventEmitter to listen to
+ * @param eventName Name of the event to listen to
+ * @param callback Callback to call when the event is triggered
+ */
+export function useListener(
+  eventEmitter: EventEmitter,
```

Comment:
Could this be fetched from a context too? Not trying to overuse contexts here, but a lot of this feels like the right use for them. We have hooks being used in components which need the state from a few levels up in the tree. And a lot of these are just using the exact same pattern. Pretty much all calls to this hook are either `useListener(goldenLayout, ...)` or `useListener(goldenLayout.eventHub, ...)`.

Maybe a `useGoldenLayoutListener` and `useListener`? The cases with `useListener(goldenLayout` are specifically for golden layout events like panel picked up or panel dropped. The others are arbitrary events we define and just plumb through golden layout

Reply:
Yup context would make sense here

Reply:
We'll address contexts in a follow-up PR, this PR shouldn't be breaking and changing to contexts would be.